### DISCUSSION TITLE
ERROR: Invalid requirement: 'torch>=2+cpu': Local version label can only be used with `==` or `!=` operators     torch>=2+cpu

### DISCUSSION THREAD

**Discussion by FeiJieBeiBei** [open-edge-platform/anomalib#2206] [Q&A]
Hello!
I tried installing with > anomalib install command but got titled error. Does anyone know how to address this? Thanks!

Installation list: ['pytest', 'coverage', 'openvino>=2024.0', 'imgaug==0.4.0', 'notebook',      install.py:73
           'pandas>=1.1.0', 'pytest-sugar', 'onnx>=1.16.0', 'pytest-xdist', 'sphinx-copybutton', 'tox',
           'wandb<=0.15.9,>=0.12.17', 'myst-parser', 'lightning>=2.2', 'gitpython', 'nncf>=2.10.0',
           'pytest-cov', 'pytest-mock', 'kornia>=0.6.6', 'jsonargparse>=4.27.7', 'freia>=0.2',
           'matplotlib>=3.4.3', 'anomalib', 'opencv-python>=4.5.3.56', 'timm<=1.0.3,>=0.5.4',
           'docstring-parser', 'rich>=13.5.2', 'av>=10.0.0', 'open-clip-torch>=2.23.0', 'rich-argparse',
           'tensorboard', 'sphinx-book-theme', 'pre-commit', 'sphinx-design', 'anomalib', 'sphinx<8.0',
           'ipykernel', 'einops>=0.3.2', 'mlflow>=1.0.0', 'ipywidgets', 'nbsphinx', 'torchmetrics>=1.3.2',
           'pandoc', 'comet-ml>=3.31.7', 'sphinx-autodoc-typehints', 'gradio>=4', 'omegaconf>=2.1.1',
           '--extra-index-url', 'https://download.pytorch.org/whl/cpu', 'torch>=2+cpu',
           'torchvision>=0.16.2']
[07/22/24 16:57:29] Invalid requirement: 'torch>=2+cpu': Local version label can only be used with `==` or `!=`
                    operators
                        torch>=2+cpu

**Comment by samet-akcay** [comment]
It is an issue reported here.
https://github.com/openvinotoolkit/anomalib/issues/2156

If you revert your pip to 24.0 it would be solved for now. We'll propose a fix to this.

**Reply by FeiJieBeiBei** [reply]
Thank you! Sorry another new issue popped up after get pip version back to 24.0, if you can kindly help:

[10:01:45] Installation list: ['jsonargparse>=4.27.7', 'matplotlib>=3.4.3', 'open-clip-torch>=2.23.0',     install.py:73
           'mlflow>=1.0.0', 'ipywidgets', 'anomalib', 'sphinx-design', 'pytest-cov', 'coverage',
           'tensorboard', 'opencv-python>=4.5.3.56', 'omegaconf>=2.1.1', 'pytest-sugar', 'pytest-mock',
           'torchmetrics>=1.3.2', 'rich-argparse', 'lightning>=2.2', 'kornia>=0.6.6', 'av>=10.0.0',
           'pandoc', 'timm<=1.0.3,>=0.5.4', 'nncf>=2.10.0', 'myst-parser', 'gradio>=4', 'notebook',
           'onnx>=1.16.0', 'anomalib', 'pandas>=1.1.0', 'pre-commit', 'tox', 'ipykernel',
           'wandb<=0.15.9,>=0.12.17', 'imgaug==0.4.0', 'sphinx<8.0', 'openvino>=2024.0',
           'docstring-parser', 'pytest-xdist', 'einops>=0.3.2', 'nbsphinx', 'gitpython',
           'sphinx-book-theme', 'sphinx-autodoc-typehints', 'freia>=0.2', 'pytest', 'rich>=13.5.2',
           'sphinx-copybutton', 'comet-ml>=3.31.7', '--extra-index-url',
           'https://download.pytorch.org/whl/cpu', 'torch>=2+cpu', 'torchvision>=0.16.2']
[07/23/24 10:03:09] Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection
                    broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at
                    0x000001501A0EA2A0>, 'Connection to download.pytorch.org timed out. (connect timeout=60.0)')':
                    /whl/cpu/matplotlib/
⠇ Installing packages...  This may take a few minutes.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x000001501A0EA2A0>, 'Connection to download.p[07/23/24 10:04:34] Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection
                    broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at
                    0x0000015018C04E00>, 'Connection to download.pytorch.org timed out. (connect timeout=60.0)')':
                    /whl/cpu/matplotlib/
